### PR TITLE
Allow esc to Close Marketplace Dialog

### PR DESCRIPTION
**Pull request by kingthorin**
Add handling to the marketplace dialog so that it can be closed with the
escape key.

### CODE REVIEW

File: src/org/zaproxy/zap/extension/autoupdate/ManageAddOnsDialog.java [hunk 1]
```diff
@@ -159,6 +164,18 @@ private void initialize() {
         	this.setSize(700, 500);
         }
         state = State.IDLE;
+        
+        //  Handle escape key to close the dialog
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/autoupdate/ManageAddOnsDialog.java]: Spaces at the end of the line could be removed.

File: src/org/zaproxy/zap/extension/autoupdate/ManageAddOnsDialog.java [hunk 2]
```diff
@@ -159,6 +164,18 @@ private void initialize() {
         	this.setSize(700, 500);
         }
         state = State.IDLE;
+        
+        //  Handle escape key to close the dialog    
+        KeyStroke escape = KeyStroke.getKeyStroke(KeyEvent.VK_ESCAPE, 0, false);
+        AbstractAction escapeAction = new AbstractAction() {
+			private static final long serialVersionUID = 3516424501887406165L;
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/autoupdate/ManageAddOnsDialog.java]: Could use spaces for indentation (as surrounding code). Same for some following statements.

File: src/org/zaproxy/zap/extension/autoupdate/ManageAddOnsDialog.java (hunk 1 above)
Comment (kingthorin) [inline src/org/zaproxy/zap/extension/autoupdate/ManageAddOnsDialog.java]: Done

File: src/org/zaproxy/zap/extension/autoupdate/ManageAddOnsDialog.java (hunk 2 above)
Comment (kingthorin) [inline src/org/zaproxy/zap/extension/autoupdate/ManageAddOnsDialog.java]: Should match now

**Comment by thc202** [comment]
Thanks!

**Pull request merged**